src/reachability/utils/reachability.jl

```diff
@@ -3,12 +3,51 @@
 # Performs layer-by-layer propagation
 # It is called by all solvers under reachability
 # TODO: also called by ReluVal and FastLin, so move to general utils (or to network.jl)
-function forward_network(solver, nnet::Network, input)
-    reach = input
+function forward_network(solver::Solver, nnet::Network, input;
+                         collect=false, before_act=false, overapprox=false, T=Any)
```

Comment:
Instead of defaulting to T = Any maybe we should implement a way of inferring the return type of forward_layer? At a low level, a lot of LazySets function are type unstable in order to return e.g. EmptySet vs HPolytope, but at the level of forward_layer we are generally careful with out returns, so we might have decent type stability we can use with function like Base.return_types or Base._return_type. For one thing, this avoids the ugly T = SymbolicInteralGradient in neurify.